lib/std/os/linux/start_pie.zig

```diff
@@ -266,4 +266,28 @@ pub fn relocate(phdrs: []elf.Phdr) void {
             @as(*usize, @ptrFromInt(base_addr + r.r_offset)).* = base_addr + @as(usize, @bitCast(r.r_addend));
         }
     }
+
+    const relr = sorted_dynv[elf.DT_RELR];
+    if (relr != 0) {
+        const relrs = @call(.always_inline, std.mem.bytesAsSlice, .{
+            elf.Relr,
+            @as([*]u8, @ptrFromInt(base_addr + relr))[0..sorted_dynv[elf.DT_RELRSZ]],
+        });
+        var current: [*]usize = undefined;
+        for (relrs) |r| {
+            if ((r & 1) == 0) {
+                current = @ptrFromInt(base_addr + r);
+                current[0] += base_addr;
```

Comment:
As I worked on the relocation stuff in this file, I did start to wonder: Should these actually be done with wrapping arithmetic in general?

Reply:
No, I don't think so. This would never overflow unless a bug.